report_format_option/models/ir_actions_report.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 Quartile Limited
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class IrActionReport(models.Model):
+    _inherit = "ir.actions.report"
+
+    encoding = fields.Char(
+        help="Encoding to be applied to the generated CSV file. e.g. cp932"
+    )
+    encode_error_handling = fields.Selection(
+        selection=[("ignore", "Ignore"), ("replace", "Replace")],
+        help="If nothing is selected, CSV export will fail with an error message when "
+        "there is a character that fail to be encoded.",
+    )
+    show_encoding = fields.Boolean(
+        compute="_compute_show_encoding",
+        help="Technical field to control the visibility of the encoding field.",
+    )
+    line_ending = fields.Selection(
+        [("LF", "LF (\\n)"), ("CRLF", "CRLF (\\r\\n)"), ("CR", "CR (\\r)")],
+        default="LF",
+        help="Select the type of line ending.",
+    )
+    show_line_ending = fields.Boolean(
+        compute="_compute_show_line_ending",
+        help="Technical field to control the visibility of the line ending field.",
+    )
+
+    def _compute_show_encoding(self):
+        """Extend this method to show the encoding field in the report form."""
+        for report in self:
+            report.show_encoding = False
+            if report.report_type == "qweb-text":
+                report.show_encoding = True
+
+    def _compute_show_line_ending(self):
+        """Extend this method to show the line ending field in the report form."""
+        for report in self:
+            report.show_line_ending = False
+            if report.report_type == "qweb-text":
+                report.show_line_ending = True
+
+    @api.model
+    def _render_qweb_text(self, report_ref, docids, data=None):
+        content, content_type = super()._render_qweb_text(report_ref, docids, data)
+        report = self._get_report(report_ref)
+        # If specific encoding is set on the report, use it; otherwise, fallback to utf-8
+        encoding = report.encoding or "utf-8"
+        # Decode content assuming it is 'utf-8' to work with it as a string
+        content_str = content.decode("utf-8")
+        # Adjust line endings if specified, while content is still a string
+        line_ending = report.line_ending or ""
+        if line_ending == "CRLF":
+            content_str = content_str.replace("\n", "\r\n")
+        elif report.line_ending == "CR":
+            content_str = content_str.replace("\n", "\r")
+        # Then encode with the desired encoding
+        encode_options = {}
+        if report.encode_error_handling:
+            encode_options["errors"] = report.encode_error_handling
+        # Re-encode the content with or without the errors parameter
+        content_encoded = content_str.encode(encoding, **encode_options)
+        return content_encoded, content_type
```

Comment:
```suggestion
        if not report.encoding and not report.line_ending:
            return content, content_type
        report = self._get_report(report_ref)
        content_str = content.decode("utf-8")
        if report.line_ending == "crlf":
            content_str = content_str.replace("\n", "\r\n")
        elif report.line_ending == "cr":
            content_str = content_str.replace("\n", "\r")
        # If specific encoding is set on the report, use it; otherwise, fallback to utf-8
        encoding = report.encoding or "utf-8"
        encode_options = {}
        if report.encode_error_handling:
            encode_options["errors"] = report.encode_error_handling
        content = content_str.encode(encoding, **encode_options)
        return content, content_type
```